jvm/tests/Language/JavaSpec.hs

```diff
@@ -90,3 +95,21 @@ spec = around_ (runInBoundThread . runInAttachedThread) $ do
         let i = maxBound :: Int32
         j <- new i :: IO (J ('Class "java.lang.Integer"))
         reify (unsafeCast j) `shouldReturn` (fromIntegral i :: Int64)
+
+      it "correctly encodes characters outside ASCII range" $ do
+        let incorrect :: Text = "आपकी उम्र लंबी हो और आप समृद्ध बने 👋"
+        let correct = Text.replace "👋" "🖖" incorrect
+        hello <- reflect ("👋" :: Text)
+        spock <- reflect ("🖖" :: Text)
+        jincorrect <- reflect incorrect
+        jcorrect :: JString <- call jincorrect "replaceFirst" hello spock
+        reify jcorrect `shouldReturn` correct
+
+    describe "strings" $ do
+      -- It is known that the current implementation of JNI.String does not support embedded NULL
+      prop "correctly convert back and forth from Prelude.String to JNI.String" $
+        \s -> (notElem '\NUL' s) ==> (s === (JNI.toChars . JNI.fromChars) s)
+      prop "correctly convert back and forth from JNI.String to Prelude.String" $
+        \t -> (not $ isInfixOf "\NUL" t) ==>
+          let s = JNI.fromByteString $ encodeUtf8 t
+          in s === (JNI.fromChars . JNI.toChars) s
```

Comment:
This is testing the utf8 encoding and decoding of ByteStrings more than JNI.String itself. Is it worth of CI time in inline-java?
Maybe testing some cases as in the tests for reify/reflect would suffice.

Reply:
The time cost of these tests is very light: the whole test suite for `jvm` runs in less than 0.1 second